lib/bunyan.js

```diff
@@ -1339,7 +1380,13 @@ RotatingFileStream.prototype.write = function write(s) {
         this.rotQueue.push(s);
         return false;
     } else {
-        return this.stream.write(s);
+        if ((this.alreadyWritten + this.stream.bytesWritten) > this.rotateFileSize) {
```

Comment:
By checking the file and rotating immediately, we lose the log message that caused the rotation